common-core-open/src/main/java/edu/isi/nlp/files/KeyValueSources.java

```diff
@@ -92,4 +93,26 @@ private KeyValueSources() {
     }
     return new ZipKeyValueSource(zipFile, ret.build());
   }
+
+  @Nonnull
+  public static ImmutableKeyValueSource<Symbol, ByteSource> fromVistaUtilsZipKeyValueSource(
+      final ZipFile zipFile) {
+    // read "__keys" file
+    ZipEntry keysEntry = zipFile.getEntry(("__keys"));
+    String keysString = null;
+    try {
+      keysString = ZipFiles.entryAsString(zipFile, keysEntry, StandardCharsets.US_ASCII);
+    } catch (IOException e) {
+      e.printStackTrace();
+    }
+    String[] keysArray = keysString.split("\n");
```

Comment:
nit: You can simplify this to

```java
List<String> keys = Splitter.on("\n").splitToList(keysString);
ImmutableMap<Symbol, String> keyMap = Maps.uniqueIndex(
    keys, key -> Symbol(key))
   ```

Reply:
thanks -- keep tips like this coming :)

Reply:
Experience-sharing like that is a good 1/3 of what code review is for :-)

Reply:
And to improve my own little snippet a bit, `keyToZipEntryName` would be a clearer name than `keyMap`.